File: generated-code/php/codecraft/FileReadWrite/Model/Entity.php
<?php

namespace Model {
    require_once 'Model/EntityType.php';
    require_once 'Stream.php';
    require_once 'Vec2Int.php';

    /**
     * Game entity
     */
    class Entity
    {
        /**
         * Entity's ID. Unique for each entity
         */
        public int $id;
        /**
         * Entity's owner player ID, if owned by a player
         */
        public ?int $playerId;
        /**
         * Entity's type
         */
        public int $entityType;
        /**
         * Entity's position (corner with minimal coordinates)
         */
        public \Vec2Int $position;
        /**
         * Current health
         */
        public int $health;
        /**
         * If entity is active, it can perform actions
         */
        public bool $active;
    
        function __construct(int $id, ?int $playerId, int $entityType, \Vec2Int $position, int $health, bool $active)
        {
            $this->id = $id;
            $this->playerId = $playerId;
            $this->entityType = $entityType;
            $this->position = $position;
            $this->health = $health;
            $this->active = $active;
        }
    
        /**
         * Read Entity from input stream
         */
        public static function readFrom(\InputStream $stream): Entity
        {
            $id = $stream->readInt32();
            if ($stream->readBool()) {
                $playerId = $stream->readInt32();
            } else {
                $playerId = NULL;
            }
            $entityType = \Model\EntityType::readFrom($stream);
            $position = \Vec2Int::readFrom($stream);
            $health = $stream->readInt32();
            $active = $stream->readBool();
            return new Entity($id, $playerId, $entityType, $position, $health, $active);
        }
        
        /**
         * Write Entity to output stream
         */
        public function writeTo(\OutputStream $stream): void
        {
            $stream->writeInt32($this->id);
            if (is_null($this->playerId)) {
                $stream->writeBool(false);
            } else {
                $stream->writeBool(true);
                $stream->writeInt32($this->playerId);
            }
            $stream->writeInt32($this->entityType);
            $this->position->writeTo($stream);
            $stream->writeInt32($this->health);
            $stream->writeBool($this->active);
        }
    }
}

File: generated-code/python/codecraft/FileReadWrite/model/entity.py
from model.entity_type import EntityType
from stream_wrapper import StreamWrapper
from typing import Optional
from vec2_int import Vec2Int

class Entity:
    """Game entity"""

    __slots__ = ("id","player_id","entity_type","position","health","active",)

    id: int
    player_id: Optional[int]
    entity_type: EntityType
    position: Vec2Int
    health: int
    active: bool

    def __init__(self, id: int, player_id: Optional[int], entity_type: EntityType, position: Vec2Int, health: int, active: bool):
        self.id = id
        """Entity's ID. Unique for each entity"""
        self.player_id = player_id
        """Entity's owner player ID, if owned by a player"""
        self.entity_type = entity_type
        """Entity's type"""
        self.position = position
        """Entity's position (corner with minimal coordinates)"""
        self.health = health
        """Current health"""
        self.active = active
        """If entity is active, it can perform actions"""

    @staticmethod
    def read_from(stream: StreamWrapper) -> "Entity":
        """Read Entity from input stream
        """
        id = stream.read_int()
        if stream.read_bool():
            player_id = stream.read_int()
        else:
            player_id = None
        entity_type = EntityType(stream.read_int())
        position = Vec2Int.read_from(stream)
        health = stream.read_int()
        active = stream.read_bool()
        return Entity(id, player_id, entity_type, position, health, active)
    
    def write_to(self, stream: StreamWrapper):
        """Write Entity to output stream
        """
        stream.write_int(self.id)
        if self.player_id is None:
            stream.write_bool(False)
        else:
            stream.write_bool(True)
            stream.write_int(self.player_id)
        stream.write_int(self.entity_type)
        self.position.write_to(stream)
        stream.write_int(self.health)
        stream.write_bool(self.active)
    
    def __repr__(self):
        return "Entity(" + \
            repr(self.id) + \
            ", " + \
            repr(self.player_id) + \
            ", " + \
            repr(self.entity_type) + \
            ", " + \
            repr(self.position) + \
            ", " + \
            repr(self.health) + \
            ", " + \
            repr(self.active) + \
            ")"